cockatrice/src/carddatabase.cpp

```diff
@@ -194,23 +196,35 @@ void SetList::markAllAsKnown()
 
 void SetList::guessSortKeys()
 {
-    // sort by release date DESC; invalid dates to the bottom.
-    QDate distantFuture(2050, 1, 1);
-    int aHundredYears = 36500;
+    defaultSort();
     for (int i = 0; i < size(); ++i) {
         CardSetPtr set = at(i);
         if (set.isNull()) {
             qDebug() << "guessSortKeys set is null";
             continue;
         }
+        set->setSortKey(i);
+    }
+}
 
-        QDate date = set->getReleaseDate();
-        if (date.isNull()) {
-            set->setSortKey(static_cast<unsigned int>(aHundredYears));
-        } else {
-            set->setSortKey(static_cast<unsigned int>(date.daysTo(distantFuture)));
+void SetList::defaultSort()
+{
+    std::sort(begin(), end(), [](const CardSetPtr &a, const CardSetPtr &b) {
+        QDate distantPast = QDate(1970, 1, 1);
+        QDate aDate = a->getReleaseDate().isValid() ? a->getReleaseDate() : distantPast;
```

Comment:
this function will get called a bunch for sorting, it might be worth optimising it just a bit by trying to return early and trying to save and reuse results of functions.

Reply:
Hmm, could you show me how best to go about this? I'm sure it's just due to my unfamiliarity with the language, but I don't see any obvious ways to optimize. As a sort function, it has to return either `true` or `false`, so I don't see any opportunity to return early, and the only functions/methods that it calls are only called once, so storing their return values doesn't look like it would help to me.

Reply:
I'm also not sure where you're trying to go with this @ebbit1q can you clarify?

Reply:
I've changed it a bit, it's much simpler now and does the same thing.